### ISSUE TITLE
Make it possible to sanitize or otherwise process values

### ISSUE THREAD

**Issue by paldepind** [Jomik/object-validator#5]
It would be nice if it was possible to pipe validated values through a function. After the validation, the given function would be applied to the result and the result would be returned.

As an example, say I'm using an [immutable list](https://github.com/funkia/list) which offers a `fromArray` function that converts arrays into immutable lists. It would then be very convenient if I could apply this conversion as the same as the validation takes place. Here is an example:

```js
const person = object({ name: string, id: number });
const schema = person.extend({
  mother: person,
  father: person,
  children: array(person).thru(fromArray)
});
```

It would also be nice if this processing could come _first_ in the validation as a way to sanitize input.

```js
const person = object({
  name: string
  // Buggy API users sends numbers as strings :'(
  age: thru((n) => parseInt(n, 10)).number.gt(18)
});
```

The name `thru` is inspired by [this `thru`](https://github.com/cujojs/most/blob/master/docs/api.md#thru). Other ideas for a name: `process`, `convert`, `sanitize`, `transform`.

**Comment by Jomik** [comment]
Have to make a small adjustment:
```ts
const person = object({
  name: string
  // Buggy API users sends numbers as strings :'(
  age: thru((n) => parseInt(n, 10)).and(number.gt(18))
});
```
As I can't properly return a `Type` from `thru`, without messing up the methods on all the other classes as they inherit from `BaseType`.
I am pondering if it should be necessary to preprocess - as you don't actually know what you can call on the value, so your actions are quite limited. Committing this for now.

**Comment by Jomik** [comment]
Fixed in #7
